### ISSUE TITLE
MorphOne on Interface/Union returns faulty data

### ISSUE THREAD

**Issue by Gaia-Nutrition** [nuwave/lighthouse#1034]
**Describe the bug**
When I try to return data from a `MorphOne` relation using an Interface, the returned data does not match the correct Eloquent model

**Expected behavior/Solution**
Given the following schema:
```graphql
interface Person {
    id: ID!
    user: User! @morphOne
}

union Creator = Employee | Contractor

type User {
    id: ID!
    name: String!
    email: String!
}

type Employee implements Person {
    id: ID!
    position: String!
    user: User! @morphOne
}

type Contractor implements Person {
    id: ID!
    position: String!
    user: User! @morphOne
}

type Color {
    id: ID!
    name: String!
    creator: Creator @morphTo
}

type Query {
    colors: [Color!]! @all
}
```
If I then execute this query:
```graphql
colors {
    id
    name
    creator {
        ... on Person {
            __typename
            id
            user {
                id
                name
                email
            }
        }
    }
}
```
I should be getting the correct `User` for each `user` `MorphOne` relation inside the `creator` union.
However, the same `User` is being returned for **both** `creator`s

**Steps to reproduce**
I've made a fork with a failing test to illustrate this
1. Clone the fork from https://github.com/Gaia-Nutrition/lighthouse
2. Install `docker-compose up -d`
3. Install composer and run test
  - Makefile was not working for me so:
  - Install composer dependencies `docker-compose exec php composer install`
  - Run the failing test: `docker-compose exec php composer test:bug`

**Output/Logs**
Example failing test output:
<details><summary>Click to expand</summary>

```
Failed asserting that an array has the subset Array &0 (
    'data' => Array &1 (
        'colors' => Array &2 (
            0 => Array &3 (
                'id' => '1'
                'name' => 'Jamir Walker'
                'creator' => Array &4 (
                    '__typename' => 'Employee'
                    'id' => '1'
                    'user' => Array &5 (
                        'id' => '1'
                        'name' => 'Alexanne Schroeder'
                        'email' => 'bayer.lacey@example.com'
                    )
                )
            )
            1 => Array &6 (
                'id' => '2'
                'name' => 'Brady Lebsack'
                'creator' => Array &7 (
                    '__typename' => 'Contractor'
                    'id' => '1'
                    'user' => Array &8 (
                        'id' => '2'
                        'name' => 'Ottilie Windler'
                        'email' => 'lincoln70@example.com'
                    )
                )
            )
        )
    )
).
--- Expected
+++ Actual
@@ @@
           'id' => '1',
           'user' =>
           array (
-            'id' => '1',
-            'name' => 'Alexanne Schroeder',
-            'email' => 'bayer.lacey@example.com',
+            'id' => '2',
+            'name' => 'Ottilie Windler',
+            'email' => 'lincoln70@example.com',
           ),
         ),
       ),
```

</details></br>

**Environment**

Lighthouse Version: master
Laravel Version: 6.x

**Comment by spawnia** [comment]
@Gaia-Nutrition thanks for the detailed report.

Can you open a PR with your failing test case? That will allow us to see the code and the failing test run in Travis.

**Comment by spawnia** [comment]
As a temporary workaround for this issue, you can disable batch loading for relations through a new config setting: https://github.com/nuwave/lighthouse/pull/1059

**Comment by spawnia** [comment]
Cleanly fixed as of #1035
